internal/pkg/source/source.go

```diff
@@ -466,6 +470,30 @@ func isSourceType(c classifier, t types.Type) bool {
 	}
 }
 
+// A type can be tainted by a call if it is itself a pointer or pointer-like type (according to
+// pointer.CanPoint), or it is an array/struct that holds an element that can be tainted by
+// a call.
+func canBeTaintedByCall(t types.Type) bool {
+	if pointer.CanPoint(t) {
+		return true
+	}
+
+	switch tt := t.(type) {
+	case *types.Array:
+		return canBeTaintedByCall(tt.Elem())
+
+	case *types.Struct:
+		for i := 0; i < tt.NumFields(); i++ {
+			if canBeTaintedByCall(tt.Field(i).Type()) {
+				return true
+			}
```

Comment:
My first impression was that this could potentially be infinitely recursive.

Mulling it over, you aren't actually able to have a non-reference recursive struct, because you'd get a build failure.  You'd only be able to recurse if a field of `type Foo struct { ...` were, e.g., `*Foo` or `[]Foo`, but not `Foo` or `[2]Foo`.

All that to say, it might be worth a comment here to explain why we aren't actually worried about the recursive step here.